peer/network.go

```diff
@@ -190,7 +190,7 @@ func (n *network) sendAppRequest(nodeID ids.NodeID, request []byte, responseHand
 
 	// Send app request to [nodeID].
 	// On failure, release the slot from [activeAppRequests] and delete request from [outstandingRequestHandlers]
-	if err := n.appSender.SendAppRequest(context.TODO(), nodeIDs, requestID, request); err != nil {
+	if err := n.appSender.SendAppRequest(ctx, nodeIDs, requestID, request); err != nil {
```

Comment:
The `common.AppSender` interface says that a non-nil error should be treated as fatal but we currently don't honor that. Can you add a TODO for us to do that pls?

Reply:
thanks